src/irr/asset/CImageWriterJPG.cpp

```diff
@@ -98,16 +96,76 @@ static void jpeg_file_dest(j_compress_ptr cinfo, io::IWriteFile* file)
 	dest->file = file;
 }
 
+template<asset::E_FORMAT outFormat>
+core::smart_refctd_ptr<asset::ICPUImage> getJPGConvertedOutput(const asset::ICPUImage* image)
+{
+	static_assert(!asset::isBlockCompressionFormat<outFormat>(), "Only non BC formats supported!");
```

Comment:
seems like PNG and JPG writer could use the same function.

I have the exact same comment here:
- don't convert if formats match, use flatten instead
- take imageview as input
- get rid of `getTopImageDataForCommonWriting` you could do this all in here

Reply:
also don't convert the entire, WHOLE image, just the miplevel+arraylayer you're actually doing to write out ;)